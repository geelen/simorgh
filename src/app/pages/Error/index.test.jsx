import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import ArticleContainer from './index';

// explicitly ignore console.log errors for Article/index:getInitialProps() error logging
global.console.log = jest.fn();

const defaultProps = {
  isAmp: false,
  pageType: 'error',
  status: 404,
  service: 'news',
};

jest.mock('../../containers/PageHandlers/withPageWrapper', () => Component => {
  const PageWrapperContainer = props => (
    <div id="PageWrapperContainer">
      <Component {...props} />
    </div>
  );

  return PageWrapperContainer;
});

jest.mock('../../containers/PageHandlers/withLoading', () => Component => {
  const LoadingContainer = props => (
    <div id="LoadingContainer">
      <Component {...props} />
    </div>
  );

  return LoadingContainer;
});

jest.mock('../../containers/PageHandlers/withContexts', () => Component => {
  const ErrorContainer = props => (
    <div id="ContextsContainer">
      <Component {...props} />
    </div>
  );

  return ErrorContainer;
});
jest.mock('../../containers/ErrorMain', () => {
  const ErrorMain = () => <div>ErrorMain</div>;

  return ErrorMain;
});

describe('ErrorContainer', () => {
  shouldMatchSnapshot(
    'should compose ErrorContainer with the Page Handler in the correct order',
    <ArticleContainer {...defaultProps} />,
  );
});
