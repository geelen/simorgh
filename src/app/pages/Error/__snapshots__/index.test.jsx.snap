// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorContainer should compose ErrorContainer with the Page Handler in the correct order 1`] = `
<div
  id="ContextsContainer"
>
  <div
    id="PageWrapperContainer"
  >
    <div
      id="LoadingContainer"
    >
      <div>
        ErrorMain
      </div>
    </div>
  </div>
</div>
`;
